File: typescript/src/cards/flashcard.ts

interface FlashCard {
  getQuestion: () => string
  getAnswer: () => string

  /**
   * Checks whether the provided response matches the target, ignoring minor issues in capitalization and whitespace.
      * @param response The user-provided response.
      * @return {@code true} if the definition matches the response.
      */
  checkSuccess: (response: string) => boolean

  toString: () => string
}

/**
 * create a new flashcard with a title and a definition
 * @param title
 * @param definition
 * @returns new flash card
 */
function newFlashCard (question: string, answer: string): FlashCard {
  return {
    getQuestion: function (): string { return question },
    getAnswer: function (): string { return answer },
    checkSuccess: function (response: string): boolean {
      return answer.trim().toLowerCase() === response.trim().toLowerCase()
    },
    toString: () => {
      return 'FlashCard[' + question + ', ' + answer + ']'
    }
  }
}

export { FlashCard, newFlashCard }

File: java/src/main/java/edu/cmu/cs214/hw1/cards/FlashCard.java
package edu.cmu.cs214.hw1.cards;

public record FlashCard(String question, String answer) {

    /**
     * Checks whether the provided response matches the target, ignoring minor issues in capitalization and whitespace.
     *
     * @param response The user-provided response.
     * @return {@code true} if the target matches the response.
     */
    public boolean checkSuccess(String response) {
        // By default, implement a lenient (whitespace- and case-insensitive) String equality check.
        return this.answer.trim().equalsIgnoreCase(response.trim());
    }


}
